### PR TITLE
Attach/Detach Google Cloud disks with pre/post-processors

### PR DESCRIPTION
**Pull request by aarontp**
Note:  A lot of this code comes from work from jberggren@ in another repository.  I'm not adding tests here because this code will likely be changed to use a separate library later on.

### CODE REVIEW

File: turbinia/processors/google_cloud.py [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+      else:
+        result = service.globalOperations().get(
+            project=self.project_id, operation=operation[u'name']).execute()
+
+      if u'error' in result:
+        raise TurbiniaException(result[u'error'])
+
+      if not block or result[u'status'] == u'DONE':
+        return result
+      time.sleep(1)  # Seconds between requests
```
Comment (berggren) [inline turbinia/processors/google_cloud.py]: I wonder if this is a bit aggressive, maybe do 5sec at least?

File: turbinia/processors/google_cloud.py [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+      A Google Compute Engine service object.
+    """
+    return self._CreateService(u'compute', self.COMPUTE_ENGINE_API_VERSION)
+
+  def GceOperation(self, operation, zone=None, block=False):
+    """Convinient method for GCE operation.
+
+    Args:
+      operation: Operation to be executed.
+      zone: GCP zone to execute the operation in. None means global operation.
```
Comment (berggren) [inline turbinia/processors/google_cloud.py]: s/None/'None'/

File: turbinia/processors/google_cloud.py [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+        'autoDelete': False,
+    }
+    operation = self.project.GceApi().instances().attachDisk(
+        instance=self.name,
+        project=self.project.project_id,
+        zone=self.zone,
+        body=operation_config).execute()
+    self.project.GceOperation(operation, zone=self.zone, block=True)
+
+  def DetachDisk(self, disk):
```
Comment (berggren) [inline turbinia/processors/google_cloud.py]: heh, nice ;)

File: turbinia/processors/google_cloud.py [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+def PostprocessDetachDisk(evidence):
+  """Detaches Google Cloud Disk from an instance.
+
+  Args:
+    evidence: A turbinia.evidence.GoogleCloudProject object.
+  """
+  if evidence.local_path:
+    path = evidence.local_path
+  else:
+    path = u'/dev/disk/by-id/google-' + evidence.disk_name
```
Comment (berggren) [inline turbinia/processors/google_cloud.py]: Use format() instead:
u'/dev/disk/by-id/google-{0:s}'.format(evidence.disk_name)

File: turbinia/processors/google_cloud.py [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+  return instance
+
+
+def PreprocessAttachDisk(evidence):
+  """Attaches Google Cloud Disk to an instance.
+
+  Args:
+    evidence: A turbinia.evidence.GoogleCloudProject object.
+  """
+  path = u'/dev/disk/by-id/google-' + evidence.disk_name
```
Comment (berggren) [inline turbinia/processors/google_cloud.py]: Use format() instead:
u'/dev/disk/by-id/google-{0:s}'.format(evidence.disk_name)

File: turbinia/processors/google_cloud.py [hunk 6]
```diff
@@ -0,0 +1,412 @@
...
+            project=self.project_id, zone=zone,
+            operation=operation[u'name']).execute()
+      else:
+        result = service.globalOperations().get(
+            project=self.project_id, operation=operation[u'name']).execute()
+
+      if u'error' in result:
+        raise TurbiniaException(result[u'error'])
+
+      if not block or result[u'status'] == u'DONE':
```
Comment (berggren) [inline turbinia/processors/google_cloud.py]: Can we get something else than "DONE" here?

Comment (aarontp) [inline turbinia/processors/google_cloud.py]: Not sure what you mean in this case.  This is something set by the API call, and is just compared here.  Let me know if there's something specific you wanted me to change.

File: turbinia/processors/google_cloud.py (hunk 1 above)
Comment (aarontp) [inline turbinia/processors/google_cloud.py]: Done.

File: turbinia/processors/google_cloud.py (hunk 2 above)
Comment (aarontp) [inline turbinia/processors/google_cloud.py]: Done.

File: turbinia/processors/google_cloud.py (hunk 5 above)
Comment (aarontp) [inline turbinia/processors/google_cloud.py]: Done.

File: turbinia/processors/google_cloud.py (hunk 4 above)
Comment (aarontp) [inline turbinia/processors/google_cloud.py]: Done.

**Comment by aarontp** [comment]
Thanks!

**Pull request merged**